features/step_definitions/applications/applications_steps.rb

```diff
@@ -1,103 +1,106 @@
+Given "{buyer} has {int} application(s)" do |buyer, number|
   plan = buyer.provider_account.first_service!.plans.default
 
   buyer.bought_cinstances.destroy_all
 
   number.to_i.times do |index|
-    FactoryBot.create(:cinstance, :user_account => buyer,
-                        :plan         => plan,
-                        :name         => "App #{index + 1}",
-                        :description  => "Yet another app")
+    FactoryBot.create(:cinstance, user_account: buyer,
+                                  plan: plan,
+                                  name: "App #{index + 1}",
+                                  description: 'Yet another app')
   end
 end
 
-Given /^the (provider ".+?") has the following applications:$/ do |provider, table|
+Given "the {provider} has the following applications:" do |provider, table|
   table.hashes.each do |row|
-    assert provider.application_plans.include?(row[:plan]) if row[:plan]
-    FactoryBot.create :cinstance, :user_account => row[:buyer],
-                        :plan => row[:plan],
-                        :name => row[:name],
-                        :description => row[:description] || "Description"
+    assert provider.application_plans.find_by(name: row['Plan']) if row['Plan']
+    FactoryBot.create :cinstance, user_account: Account.buyers.find_by(name: row['Buyer']),
+                                  plan: ApplicationPlan.find_by(name: row['Plan']),
+                                  name: row['Name'],
+                                  description: row['Description'] || 'Description'
   end
 end
 
-Given /^(application "[^"]*") is suspended$/ do |application|
+Given "{application} is suspended" do |application|
   application.suspend!
 end
 
-Given /^(application "[^"]*") has extra field "([^"]*)" blank$/ do |app, attr|
-  app.update_attribute(:extra_fields, {attr => nil})
+Given "{application} has extra field {string} blank" do |app, attr|
+  app.update!(extra_fields: { attr => nil })
 end
 
-Given /^(buyer "[^"]*") has no live applications$/ do |buyer|
-  buyer.bought_cinstances.map &:suspend!
+Given "{buyer} has no live applications" do |buyer|
+  buyer.bought_cinstances.map(&:suspend!)
 end
 
-When /^I follow "([^"]*)" for (application "[^"]*")$/ do |label, application|
+When "I follow {string} for {application}" do |label, application|
   step %(I follow "#{label}" within "#application_#{application.id}")
 end
 
-
-Then /^(application "[^"]*") should be live$/ do |application|
+Then "{application} should be live" do |application|
   assert application.live?
 end
 
-Then /^(application "[^"]*") should be suspended$/ do |application|
+Then "{application} should be suspended" do |application|
   assert application.suspended?
 end
 
-Then /^I should see that (application "[^"]*") is (live|suspended)$/ do |application, state|
-  step %(I should see "#{state.titleize}" within "#cinstance_state")
+Then "I should see that {application} is {live_state}" do |application, state|
```

Comment:
This is great. I don't request any change. But I'm still thinking. `live_state` sounds as a reasonable name. Still step user needs to go to transformation definitions to see what are the available state strings and returned data type. While original regular expression had everything obvious. I'm wondering whether the new step expressions are really that great of an idea... or maybe there's something I'm missing. e.g. IDE support might be a reasonable solution.